test/parse/TestPythonParser.cpp

```diff
@@ -0,0 +1,101 @@
+#include <array>
+#include <boost/test/tools/old/interface.hpp>
+#include <boost/test/unit_test.hpp>
+
+#include "parse/Parse.h"
+#include "parse/PythonParser.h"
+#include "universe/Effect.h"
+#include "universe/Tech.h"
+#include "universe/UnlockableItem.h"
+#include "util/Directories.h"
+#include "util/GameRules.h"
+#include "util/Pending.h"
+
+#include "ParserAppFixture.h"
+
+BOOST_FIXTURE_TEST_SUITE(TestPythonParser, ParserAppFixture)
+
+BOOST_AUTO_TEST_CASE(parse_game_rules) {
+    PythonParser parser(m_python);
+
+    Pending::Pending<GameRules> game_rules_p = Pending::Parsed(parse::game_rules, parser, m_scrpiting_dir / "game_rules.focs.py");
+    auto game_rules = *Pending::WaitForPendingUnlocked(std::move(game_rules_p));
+    BOOST_REQUIRE(!game_rules.Empty());
+    BOOST_REQUIRE(game_rules.RuleExists("RULE_HABITABLE_SIZE_MEDIUM"));
+    BOOST_REQUIRE(GameRules::Type::TOGGLE == game_rules.GetType("RULE_ENABLE_ALLIED_REPAIR"));
+}
+
+BOOST_AUTO_TEST_CASE(parse_techs) {
+    PythonParser parser(m_python);
+
+    Pending::Pending<TechManager::TechParseTuple> techs_p = Pending::StartParsing(parse::techs<TechManager::TechParseTuple>, m_scrpiting_dir / "techs");
+    auto techs_tuple = *Pending::WaitForPendingUnlocked(std::move(techs_p));
+    auto techs = std::get<0>(std::move(techs_tuple));
+    auto tech_categories = std::get<1>(std::move(techs_tuple));
+    auto categories_seen = std::get<2>(std::move(techs_tuple));
+    BOOST_REQUIRE(!techs.empty());
+    BOOST_REQUIRE(!tech_categories.empty());
+    BOOST_REQUIRE(!categories_seen.empty());
+
+    for (const auto& cat : {"PRODUCTION_CATEGORY", "GROWTH_CATEGORY"}) {
+        BOOST_REQUIRE_EQUAL(1, categories_seen.count(cat));
+        BOOST_REQUIRE_EQUAL(1, tech_categories.count(cat));
+    }
+
+    const auto cat_it = tech_categories.find("CONSTRUCTION_CATEGORY");
+    BOOST_REQUIRE(tech_categories.end() != cat_it);
```

Comment:
What is your intention about this line?  Can it be broken just by rearranging entries in the python file?

Reply:
No, it only checks if `CONSTRUCTION_CATEGORY` present in categories.